homeassistant/components/tuya/vacuum.py

```diff
@@ -147,9 +148,17 @@ def state(self) -> str | None:
             self.device.status.get(DPCode.STATUS)
         ):
             return STATE_PAUSED
+        if self.device.status.get(DPCode.FAULT):
+            return STATE_ERROR
         if not (status := self.device.status.get(DPCode.STATUS)):
             return None
-        return TUYA_STATUS_TO_HA.get(status)
+        if ha_status := TUYA_STATUS_TO_HA.get(status):
+            return ha_status
+        if "clean" in status:
+            return STATE_CLEANING
+        if "pause" in status:
+            return STATE_PAUSED
```

Comment:
This seems inefficient compared to the previous method. This should not be fuzzy.

Reply:
I haven't removed the previous method, this simply adds some extra fallback behaviour in the case where a map value is not found?

Reply:
Removed